donations/tests.py

```diff
@@ -178,7 +179,57 @@ def test_annon_donation_stripe(self):
             self.assertEqual(donations_query[0].source, 's')
             self.assertEqual(donations_query[0].amount, 15.0)
 
-    def test_donation_form(self):
+    def test_donation_form_stripe(self):
+        donations.models.DonationCampaign.objects.create(\
+                goal=200, date_start=datetime.datetime.now(), id=1)
+
+        Session = namedtuple('Session', 'id')
+        session = Session(id=1)
+        data = {
+            'amount': '0,1',
+            'show_amount': True,
+            'donation_type': '1',
+        }
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
+            mock_create.return_value = session
+            ret = self.client.get("/donations/donation-session-stripe/", data)
+            #  If GET return error 400
+            self.assertEqual(ret.status_code, 400)
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
+            mock_create.return_value = session
+            ret = self.client.post("/donations/donation-session-stripe/", data)
+            response =  ret.json()
+            # Decimals must have '.' and not ','
+            self.assertTrue('errors' in response)
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
+            mock_create.return_value = session
+            data['amount'] = '0.1'
+            ret = self.client.post("/donations/donation-session-stripe/", data)
+            response =  ret.json()
+            # amount must be greater than 1
+            self.assertTrue('errors' in response)
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
```

Comment:
you could add a check to the tests that succeed to ensure that create() was called with the correct arguments. I understand that all of the ones that fail won't call this method, right?

Reply:
Done